tests/model_registry/model_catalog/test_filter_options_endpoint.py

```diff
@@ -74,48 +80,67 @@ def test_filter_options_endpoint_validation(
-           - Execute psql commands via pod.execute()
-           - Query same tables/columns the endpoint uses
-           - Extract all distinct values for string properties: SELECT DISTINCT license FROM models;
-           - Extract min/max ranges for numeric properties: SELECT MIN(metric), MAX(metric) FROM models;
-
-        3. Compare results:
-           - API response filter values should match DB query results exactly
-           - Ensure no values are missing (comprehensive coverage)
-           - Validate across all sources, not just one
-
-        4. DB Access Pattern Example:
-           catalog_pod = get_model_catalog_pod(client, namespace)[0]
-           result = catalog_pod.execute(
-               command=["psql", "-U", "catalog_user", "-d", "catalog_db", "-c", "SELECT DISTINCT license FROM models;"],
-               container="catalog"
-           )
-
-        5. Implementation considerations:
-           - Handle different data types (strings vs arrays like tasks)
-           - Parse psql output correctly
-           - Handle null/empty values
-           - Ensure database connection credentials are available
+        This test executes the exact same SQL query the API uses and compares results
+        to catch any discrepancies between database content and API response.
+
+        Expected failure because of RHOAIENG-37069
         """
-        pytest.skip("TODO: Implement comprehensive coverage validation after backend investigation")
+        api_url = f"{model_catalog_rest_url[0]}models/filter_options"
+        LOGGER.info(f"Testing comprehensive database coverage for: {api_url}")
+
+        api_response = execute_get_command(
+            url=api_url,
+            headers=get_rest_headers(token=user_token_for_api_calls),
+        )
+
+        api_filters = api_response["filters"]
+        LOGGER.info(f"API returned {len(api_filters)} filter properties: {list(api_filters.keys())}")
+
+        postgres_pod = get_postgres_pod_in_namespace(namespace=model_registry_namespace)
+        LOGGER.info(f"Using PostgreSQL pod: {postgres_pod.name}")
+
+        db_result = postgres_pod.execute(
+            command=["psql", "-U", "catalog_user", "-d", "model_catalog", "-c", FILTER_OPTIONS_DB_QUERY],
+            container="postgresql",
+        )
+
+        db_properties = parse_psql_array_agg_output(psql_output=db_result)
+        LOGGER.info(f"Raw database query returned {len(db_properties)} properties: {list(db_properties.keys())}")
+
+        is_valid, comparison_errors = compare_filter_options_with_database(
+            api_filters=api_filters, db_properties=db_properties, excluded_fields=API_EXCLUDED_FILTER_FIELDS
+        )
+
+        if not is_valid:
+            failure_msg = "Filter options API response does not match database content"
+            failure_msg += "\nDetailed comparison errors:\n" + "\n".join(comparison_errors)
+            assert False, failure_msg
```

Comment:
why if not instead of the assert?

Reply:
just so the failure_msg formatting is a bit cleaner